File: DS/Stacks/Python/reverse_string.py
def reverseString(str1, n):
    stack = []
    for i in range(n):
        stack.append(str1[i])

    str1 = ""

    for i in range(n):
        str1 = str1 + stack.pop()

    return str1


s = "HELLO"
print(reverseString(s, len(s)))

# output: OLLEH

File: DS/Stacks/Java/reverse_string.java
import java.util.*;

public class Stacks {

    public static void main(String[] args){
        String s = "HELLO";
        Stack<Character> stack = new Stack<Character>();
        int i,n;
        n = s.length();
        for (i=0;i<n;i++){
            stack.push(s.charAt(i));
        }
        s = "";
        for(i=0;i<n;i++){
            s = s + stack.peek();
            stack.pop();
        }
        System.out.println(s);
    }
}

// output : OLLEH
